test/Turn.spec.js

```diff
@@ -0,0 +1,125 @@
+QUnit.test("Turn constructor creates a turn with the passed elements", function (assert)
+{
+    const actual = new Turn(1, 2, 3);
+	assert.deepEqual(1, actual.player);
+	assert.deepEqual(2, actual.board);
+	assert.deepEqual(3, actual.legalActions);
+});
+
+QUnit.test("Turn.EndTurn() has no default behavior (returns undefined)", function (assert)
+{
+	const actual = new Turn(1, 2, 3);
+	assert.deepEqual(undefined, actual.EndTurn());
+	/* Asserting values unchanged */
+	assert.deepEqual(1, actual.player);
+	assert.deepEqual(2, actual.board);
+	assert.deepEqual(3, actual.legalActions);
+});
+
+QUnit.test("Turn.Validate() invalidates moves for the wrong player", function(assert)
+{
+	const move = {
+		move: true,
+		capture: false,
+		source: 0
+	};
+	const board = {
+		contents: [{player: 0}, {player: 1}, {player: 1}]
+	}
+	const legalActions = {
+		move: true,
+		capture: true,
+		piece: undefined
+	};
+	const turn = new Turn(1, board, legalActions);
+	try
```

Comment:
Check out the documentation for Qunit's throw assertion: https://api.qunitjs.com/assert/throws

Reply:
I don't think this is consuming the error, and I can't find any documentation about how to make it do so. Regardless, ```assert.throws(turn.Validate(move));``` errors. Actually writing the try/catch myself does not.